File: sentimental-credit/credit.py
# TODO
N = input("Number: ")
Z = 0
P = int(N)
i=1
while(P!=0):
    t=P%10
    if(i%2!=0):
        Z+=t
    else:
        if(2*t>=10):
            Z+= 2*t % 10 + 1
        else:
            Z+= 2*t
    i+=1
    P=P//10

K = int(N[0])*10 + int(N[1])
if((Z%10==0) and ((K==34) or (K==37) or (K>=51 and K<=55) or (K>=40 and K<= 49))):
    if((K==34 or K==37) and len(N)==15):
        print("AMEX")
    elif(K>=51 and K<=55 and len(N)==16):
        print("MASTERCARD")
    elif(K>=40 and K<=49 and (len(N)==16 or len(N)==13)):
        print("VISA")
    else:
        print("INVALID")
else:
    print("INVALID")
print(Z,K,len(N))

File: credit/credit.c
#include <cs50.h>
#include <stdio.h>

int main(void)
{
    long int n;
    printf("Number: ");
    scanf("%ld",&n);
    int p=1;
    int s=0;
    int z;
    while(n!=0){
        int y=n%10;
        if(p%2!=0){
            s+=y;
        }
        else{
            if((2*y)>=10){
                s += (2*y)/10 + (2*y)%10;
            }
            else{
                s+= 2*y;
            }
        }
        if(n>=10 && n<100){
            z=n;
        }
        p++;
        n=n/10;
    }
    if((s%10==0) && ((z==34) || (z==37) || (z>=51 && z<=55) || (z>=40 && z<= 49) )){
        if((z==34) || (z==37) && (p-1)==15 ){
            printf("AMEX\n");
        }
        else if((z>=51) && (z<=55) && (p-1)==16){
            printf("MASTERCARD\n");
        }
        else if((z>=40) && (z<=49) && ((p-1)==16 || (p-1)==13) ){
            printf("VISA\n");
        }
        else{
            printf("INVALID\n");
        }
    }
    else{
        printf("INVALID\n");
    }
}